### PR TITLE
feat(argo-workflows): Add honorLabels parameter to Argo Workflows ServiceMonitor

### PR DESCRIPTION
**Pull request by venegator**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-workflows/Chart.yaml [hunk 1]
```diff
@@ -16,5 +16,7 @@ annotations:
     fingerprint: 2B8F22F57260EFA67BE1C5824B11F800CD9D2252
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
+    - kind: changed
+      description: add honorLabels config for ServiceMonitor resource
     - kind: changed
       description: add loadBalancerClass field for Service resources
```
Comment (kiblik) [inline charts/argo-workflows/Chart.yaml:22]:
```suggestion
```

File: charts/argo-workflows/templates/controller/workflow-controller-servicemonitor.yaml [hunk 2]
```diff
@@ -24,6 +24,9 @@ spec:
       metricRelabelings:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- if .Values.controller.metricsConfig.honorLabels }}
+      honorLabels: true
+      {{- end }}
```
Comment (yu-croco) [inline charts/argo-workflows/templates/controller/workflow-controller-servicemonitor.yaml:29]: nit; what about this ?

```diff
-      {{- if .Values.controller.metricsConfig.honorLabels }}
-      honorLabels: true
+      honorLabels:{{ .Values.controller.metricsConfig.honorLabels }}
-      {{- end }}
```

Comment (venegator) [reply charts/argo-workflows/templates/controller/workflow-controller-servicemonitor.yaml]: Nice one, just commited the change now

**Review by yu-croco — APPROVED** [review]
LGTM

**Review by vladlosev — APPROVED** [review]
LGTM

**Pull request merged**